### PR TITLE
osquery 2.2.1

### PR DESCRIPTION
**Pull request by ilovezfs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

CC @theopolis

### CODE REVIEW

File: Formula/osquery.rb [hunk 1]
```diff
@@ -122,9 +215,18 @@ def install
...
+      -Daws-cpp-sdk-firehose_library:FILEPATH=#{libexec}/aws-sdk-cpp/lib/libaws-cpp-sdk-firehose.a
+      -Daws-cpp-sdk-kinesis_library:FILEPATH=#{libexec}/aws-sdk-cpp/lib/libaws-cpp-sdk-kinesis.a
+      -Daws-cpp-sdk-sts_library:FILEPATH=#{libexec}/aws-sdk-cpp/lib/libaws-cpp-sdk-sts.a
+      -Dboost_filesystem-mt_library:FILEPATH=#{libexec}/boost/lib/libboost_filesystem-mt.a
+      -Dboost_regex-mt_library:FILEPATH=#{libexec}/boost/lib/libboost_regex-mt.a
+      -Dboost_system-mt_library:FILEPATH=#{libexec}/boost/lib/libboost_system-mt.a
+      -Dcppnetlib-client-connections_library:FILEPATH=#{libexec}/cpp-netlib/lib/libcppnetlib-client-connections.a
+      -Dcppnetlib-uri_library:FILEPATH=#{libexec}/cpp-netlib/lib/libcppnetlib-uri.a
+      -Dlinenoise_library:FILEPATH=#{libexec}/linenoise/lib/liblinenoise.a
+      -Dthrift_library:FILEPATH=#{libexec}/thrift/lib/libthrift.a
```
Comment (MikeMcQuaid) [inline Formula/osquery.rb]: Could probably avoid installing all the `.a`s into `libexec` as they will be statically linked happily from the `buildpath`

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/osquery.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/osquery.rb]: The test needs to be able to reference them as well.

Comment (MikeMcQuaid) [inline Formula/osquery.rb]: Cool. Might be worth considering a test that runs the application rather than building the library if possible.

**Comment by ilovezfs** [comment]
@MikeMcQuaid PR refreshed.

**Comment by theopolis** [comment]
If this is vendoring `cpp-netlib` perhaps its formula should be removed? Some of the headers it installs will not compile in downstream applications with boost 1.62+ and from what I can see only `osquery` was using it as a dependency.

**Comment by ilovezfs** [comment]
Analytics show `cpp-netlib` has only 162 installs in the last thirty days, so it's indeed pretty marginal.

**Comment by ilovezfs** [comment]
:green_apple: hurray

**Comment by ilovezfs** [comment]
Shipped.

**Comment by ilovezfs** [comment]
>If this is vendoring cpp-netlib perhaps its formula should be removed? Some of the headers it installs will not compile in downstream applications with boost 1.62+ and from what I can see only osquery was using it as a dependency.

@theopolis I've opened a PR to boneyard cpp-netlib here: https://github.com/Homebrew/homebrew-core/pull/8498.

**Pull request closed without merging**